test/Microsoft.Health.Fhir.R5.Tests.Integration/Persistence/FhirStorageTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Tests.Common;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.Integration.Persistence
+{
+    public partial class FhirStorageTests : IClassFixture<FhirStorageTestsFixture>
+    {
+        [Fact]
+        public async Task WhenUpsertingASavedResourceWithInvalidETagHeader_GivenR5Server_ThenPreconditionFailedIsThrown()
```

Comment:
We mostly have GivenX_WhenX_Then tests, though there were some the other way around. I'll made an issue to fix these